.github/workflows/build.yml

```
@@ -0,0 +1,38 @@
+name: build
+
+on: [push]
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        build_type: [Debug, Release]
+    
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Install Ubuntu Packages
+      if: matrix.os == 'ubuntu-latest'
+      shell: bash
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y libassimp-dev libglfw3-dev
+
+    - name: Install MacOS Packages
+      if: matrix.os == 'macos-latest'
+      shell: bash
+      run: |
+        brew install assimp
+        brew install glfw
+
+    - name: Generate
+      shell: bash
+      run: cmake -Bbuild -DCMAKE_BUILD_TYPE=${{ matrix.build_type }}
+      
+    - name: Build
+      shell: bash
+      run: cmake --build build/
```

```suggestion
      run: cmake --build ${{ github.workspace }}/build --config ${{ matrix.build_type }}
```
1. Same comment as earlier regarding the _build/_ folder.
2. Some CMake generators (like Xcode, Ninja Multi-Config) support multiple configurations allowing you to only select the build type at build time. Those generators ignore the specified `CMAKE_BUILD_TYPE`, and instead rely on the configuration given to `cmake --build …`; if no configuration is specified, it seems to default to a _Debug_ build.